### ISSUE TITLE
Migration example does not work

### ISSUE THREAD

**Issue by laibulle** [bitwalker/distillery#347]
### Steps to reproduce

Create a new phoenix application, configure the migrations as described here https://github.com/bitwalker/distillery/blob/master/docs/Running%20Migrations.md.

run 
```
MIX_ENV=prod mix release
PORT=8080 _build/prod/rel/haum/bin/haum migrate
```

### Verbose Logs

```
Erlang/OTP 20 [erts-9.0.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.5.1
λ ~/Projects/haum/ master* DATABASE_URL=postgres://postgres:postgres@localhost:15432/haum_dev  PORT=8080 _build/prod/rel/haum/bin/haum migrate --verbose
Loading ..
{"init terminating in do_boot",{{badmatch,{error,{"no such file or directory","nil.app"}}},[{'Elixir.Haum.ReleaseTasks',seed,0,[{file,"lib/release_tasks.ex"},{line,18}]},{init,start_em,1,[]},{init,do_boot,3,[]}]}}
init terminating in do_boot ({{badmatch,{error,{[_],[_]}}},[{Elixir.Haum.ReleaseTasks,seed,0,[{_},{_}]},{init,start_em,1,[]},{init,do_boot,3,[]}]})

Crash dump is being written to: erl_crash.dump...done
```

Paste the output of the release command you ran with the `--verbose` flag.

### Description of issue

Distillery migration do not work https://github.com/bitwalker/distillery/blob/master/docs/Running%20Migrations.md. The `myapp` var is nil.

```
defmodule Haum.ReleaseTasks do
  @start_apps [
    :crypto,
    :ssl,
    :postgrex,
    :ecto
  ]

  def myapp, do: Application.get_application(__MODULE__)

  def repos, do: Application.get_env(myapp(), :ecto_repos, [])

  def seed do
    me = myapp()

    IO.puts "Loading #{me}.."
    # Load the code for myapp, but don't start it
    :ok = Application.load(me)

    IO.puts "Starting dependencies.."
    # Start apps necessary for executing migrations
    Enum.each(@start_apps, &Application.ensure_all_started/1)

    # Start the Repo(s) for myapp
    IO.puts "Starting repos.."
    Enum.each(repos(), &(&1.start_link(pool_size: 1)))

    # Run migrations
    migrate()

    # Run seed script
    Enum.each(repos(), &run_seeds_for/1)

    # Signal shutdown
    IO.puts "Success!"
    :init.stop()
  end

  def migrate, do: Enum.each(repos(), &run_migrations_for/1)

  def priv_dir(app), do: "#{:code.priv_dir(app)}"

  defp run_migrations_for(repo) do
    app = Keyword.get(repo.config, :otp_app)
    IO.puts "Running migrations for #{app}"
    Ecto.Migrator.run(repo, migrations_path(repo), :up, all: true)
  end

  def run_seeds_for(repo) do
    # Run the seed script if it exists
    seed_script = seeds_path(repo)
    if File.exists?(seed_script) do
      IO.puts "Running seed script.."
      Code.eval_file(seed_script)
    end
  end

  def migrations_path(repo), do: priv_path_for(repo, "migrations")

  def seeds_path(repo), do: priv_path_for(repo, "seeds.exs")

  def priv_path_for(repo, filename) do
    app = Keyword.get(repo.config, :otp_app)
    repo_underscore = repo |> Module.split |> List.last |> Macro.underscore
    Path.join([priv_dir(app), repo_underscore, filename])
  end
end
```

**Comment by webdeb** [comment]
is `myapp` the name of your application?

**Comment by webdeb** [comment]
Ah, understand, you are trying to get the name of your application with `Application.get_application(__MODULE__)` hmm..

**Comment by webdeb** [comment]
found it, try the previous approach: https://github.com/bitwalker/distillery/commit/187e3ba64cb9162211959f9524f03fddd2c5954c#diff-081a77ddd39a3147d599454289f579ce

**Comment by laibulle** [comment]
Thanks for the answer.

I tried with this script 
```
defmodule Haum.ReleaseTasks do
  @start_apps [
    :crypto,
    :ssl,
    :postgrex,
    :ecto
  ]

  @myapps [
    :haum
  ]

  @repos [
    Haum.Repo
  ]

  def seed do
    # Load the code for myapp, but don't start it
    :ok = Application.load(:haum)

    IO.puts "Starting dependencies.."
    # Start apps necessary for executing migrations
    Enum.each(@start_apps, &Application.ensure_all_started/1)

    # Start the Repo(s) for myapp
    IO.puts "Starting repos.."
    Enum.each(@repos, &(&1.start_link(pool_size: 1)))
    Enum.each(@myapps, &run_migrations_for/1)

    # Signal shutdown
    IO.puts "Success!"
    :init.stop()
  end

  def priv_dir(app), do: "#{:code.priv_dir(app)}"

  defp run_migrations_for(app) do
    IO.puts "Running migrations for #{app}"
    Ecto.Migrator.run(Haum.Repo, migrations_path(app), :up, all: true)
  end

  def run_seeds_for(repo) do
    # Run the seed script if it exists
    seed_script = seeds_path(repo)
    if File.exists?(seed_script) do
      IO.puts "Running seed script.."
      Code.eval_file(seed_script)
    end
  end

  def migrations_path(repo), do: priv_path_for(repo, "migrations")

  def seeds_path(repo), do: priv_path_for(repo, "seeds.exs")

  def priv_path_for(repo, filename) do
    app = Keyword.get(repo.config, :otp_app)
    repo_underscore = repo |> Module.split |> List.last |> Macro.underscore
    Path.join([priv_dir(app), repo_underscore, filename])
  end
end
```

but I get 
```
Starting dependencies..
Starting repos..
Running migrations for haum
{"init terminating in do_boot",{undef,[{haum,config,[],[]},{'Elixir.Haum.ReleaseTasks',priv_path_for,2,[{file,"lib/release_tasks.ex"},{line,56}]},{'Elixir.Haum.ReleaseTasks',run_migrations_for,1,[{file,"lib/release_tasks.ex"},{line,39}]},{'Elixir.Enum','-each/2-lists^foreach/1-0-',2,[{file,"lib/enum.ex"},{line,675}]},{'Elixir.Enum',each,2,[{file,"lib/enum.ex"},{line,675}]},{'Elixir.Haum.ReleaseTasks',seed,0,[{file,"lib/release_tasks.ex"},{line,28}]},{init,start_em,1,[]},{init,do_boot,3,[]}]}}
init terminating in do_boot ({undef,[{haum,config,[],[]},{Elixir.Haum.ReleaseTasks,priv_path_for,2,[{_},{_}]},{Elixir.Haum.ReleaseTasks,run_migrations_for,1,[{_},{_}]},{Elixir.Enum,-each/2-lists^forea
```

**Comment by mmartinson** [comment]
Can this issue be closed?

**Comment by bitwalker** [comment]
Closed via #362
